test/async-hooks/test-async-local-storage-gcable.js

```diff
@@ -15,6 +15,7 @@ asyncLocalStorage.run({}, () => {
 
   onGC(asyncLocalStorage, { ongc: common.mustCall() });
 });
+asyncLocalStorage.disable();
```

Comment:
This is needed because `disable` clears the store only from the current `AsyncContextFrame` but after run a new `AsyncContextFrame` is created.

Reply:
Can you add a comment explaining this?

Reply:
I think we should discourage `disable`: https://github.com/nodejs/node/pull/58065

Reply:
Added a comment and a check which variant is used in the test in 7f4ed9830998653865d6697c1ba1effc4d86af61 
I agree that `disable()` should be deprecated/removed but that's for a different PR.